### PR TITLE
Adds version suffix to LogisimVersion class 

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
Makes LogisimVersion class understand optional version suffix (i.e. `3.6.0-beta1`).

### CODE REVIEW

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski , please make linter happy and ask a second reviewer for review as I'm from now one week on holidays :-)

File: src/main/java/com/cburch/logisim/LogisimVersion.java [hunk 1]
```diff
@@ -103,16 +106,27 @@ public int compareTo(LogisimVersion other) {
...
 
+  /**
+   * Returns TRUE if version is considered stable, false otherwise. Note the implementation is
+   * plaind dumb and relies on presence of version suffix (which, if not empty means non-stable
+   * release).
+   *
+   * @return
+   */
+  public boolean isStable() {
+    return "".equals("");
```
Comment (davidhutchens) [inline src/main/java/com/cburch/logisim/LogisimVersion.java:127]: I think this should be: return suffix.equals("");

Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/LogisimVersion.java:127]: Yes, that was the issue. But I fixed it already.

Comment (davidhutchens) [inline src/main/java/com/cburch/logisim/LogisimVersion.java:127]: No, you fixed the if on line 139 in the next commit. This is a return on line 127. The return still needs to be fixed.

**Review by davidhutchens — APPROVED** [review]
This looks good to me.

**Comment by MarcinOrlowski** [comment]
It appears `rpmbuild` do not like version strings with dash, i.e. `3.6.0-beta1`. It can however accept `3.6.0beta1`. DEB likes both. Needs to check mac and Windows yet but regardless their support the problem is where this needs to be addressed. Shall RPM builder just take the `-` out of version string by itself, or we make `LogisimVersion.toString()` return dashless form of version string and also make it properly parse such version strings (in current implementation `-` is actually crucial and expected separator. The fix in RPM builder is much cheaper and faster. The update of LogisimVersion class to properly parse its the correct approach. I'd fix the builder and create a ticket to update version class at some point. Also there's one potential probielem with save files? Shall we putting dash version or maybe strip dash too. So that needs works. Or we shall have suffix to display only, so saves from `3.6.0beta1` are saved as `3.6.0`

@BFH-ktt1 @davidhutchens @maehne

**Comment by davidhutchens** [comment]
As I noted in #885, the version string in the builder (as it is passed to jpackage) cannot have alphabetic characters. In the macOS build, jpackage expects it to be digits with up to two periods. I think that is consistent with version numbers as macOS expects them.

That suggests that any suffix will need to be display only.

**Comment by MarcinOrlowski** [comment]
It actually is not a problem for jbuilder. It cannot have `-`, but `3.6.0beta1' forks for both RPM and DEB packages I checked.

**Comment by MarcinOrlowski** [comment]
>  In the macOS build, jpackage expects it to be digits with up to two periods.

Can you manually force version passed to jbuilder on macOS to be say `3.6.0beta1` and tell me the results? If it fails, then how people handle `beta` or `RC` releases on macOS then?

**Pull request merged**